docs/fassets/developer-guides/12-fassets-autoredeem.mdx

```diff
@@ -0,0 +1,649 @@
+  COSTON2_FTESTXRP: "0x8b4abA9C4BD7DD961659b02129beE20c6286e17F",
+  COSTON2_OFT_ADAPTER: "0xCd3d2127935Ae82Af54Fc31cCD9D3440dbF46639",
+  COSTON2_COMPOSER: process.env.COSTON2_COMPOSER || "",
+  HYPERLIQUID_EID: EndpointId.HYPERLIQUID_V2_TESTNET,
+  EXECUTOR_GAS: 200_000,
+  BRIDGE_AMOUNT: "11", // Change this to your desired amount
+};
+```
+
+### How to Run
+
+**Run this script FIRST before attempting auto-redemption.** This gets your FXRP tokens onto Hyperliquid EVM.
+
+1. **Install Dependencies**:
+
+   ```bash
+   yarn install
+   ```
+
+2. **Configure Environment**:
+
+   ```bash
+   # .env file
+   COSTON2_RPC_URL=https://coston2-api.flare.network/ext/C/rpc
+   DEPLOYER_PRIVATE_KEY=your_private_key_here
+   COSTON2_COMPOSER=0x... # Optional for this step, required for Step 2
+   ```
+
+3. **Run the Script on Coston2**:
+
+   ```bash
+   yarn hardhat run scripts/fassets/bridgeToHyperEVM.ts --network coston2
+   ```
+
+4. **Wait for Completion**:
+   - Monitor the transaction on LayerZero Scan (link provided in output)
+   - Allow 2-5 minutes for cross-chain delivery
+   - Verify FXRP balance increased on Hyperliquid EVM before proceeding to Step 2
+
+### Expected Output
+
+```
+Using account: 0x742d35Cc6634C0532925a3b844Bc454e4438f44e
+
+📋 Bridge Details:
+From: Coston2
+To: Hyperliquid EVM Testnet
+Amount: 11.0 FXRP
+Recipient: 0x742d35Cc6634C0532925a3b844Bc454e4438f44e
+
+Your FTestXRP balance: 100.0
+
+1️⃣ Checking OFT Adapter token address...
+   OFT Adapter's inner token: 0x8b4abA9C4BD7DD961659b02129beE20c6286e17F
+   Expected token: 0x8b4abA9C4BD7DD961659b02129beE20c6286e17F
+   Match: true
+
+   Approving FTestXRP for OFT Adapter...
+   OFT Adapter address: 0xCd3d2127935Ae82Af54Fc31cCD9D3440dbF46639
+   Amount: 11.0 FXRP
+✅ OFT Adapter approved
+   Verified allowance: 22.0 FXRP
+
+3️⃣ LayerZero Fee: 0.001234 C2FLR
+
+4️⃣ Sending FXRP to Hyperliquid EVM Testnet...
+Transaction sent: 0xabc123...
+✅ Confirmed in block: 12345678
+
+🎉 Success! Your FXRP is on the way to Hyperliquid EVM Testnet!
+
+Track your transaction:
+https://testnet.layerzeroscan.com/tx/0xabc123...
+
+It may take a few minutes to arrive on Hyperliquid EVM Testnet.
+```
+
+<details>
+<summary>View `bridgeToHyperEVM.ts` source code</summary>
+
+<CodeBlock language="typescript" title="scripts/fassets/bridgeToHyperEVM.ts">
+  {bridgeToHyperEVM}
+</CodeBlock>
+
+</details>
+
+### Troubleshooting
+
+**Error: Insufficient FTestXRP balance**
+
+- Solution: Acquire FTestXRP tokens on Coston2 first
+- Token address: `0x8b4abA9C4BD7DD961659b02129beE20c6286e17F`
+
+**Error: Insufficient C2FLR for gas**
+
+- Solution: Get C2FLR from Coston2 faucet: https://faucet.flare.network/coston2
+
+**Error: Transaction reverted**
+
+- Check that the OFT Adapter address matches the FTestXRP token
+- Verify LayerZero endpoint is operational
+- Ensure gas limits are sufficient
+
+---
+
+## Auto-Redeem from Hyperliquid EVM (Step 2)
+
+### Script: `autoRedeemHyperEVM.ts`
+
+### What It Is
```

Comment:
Combine in one heading as mentioned above. Like Code Breakdown of `autoRedeemHyperEVM.ts`